contracts/modules/LoanClosingsWith.sol

```diff
@@ -145,6 +153,16 @@ contract LoanClosingsWith is LoanClosingsShared {
         //TODO should we skip this check if invoked from rollover ?
         (Loan storage loanLocal, LoanParams storage loanParamsLocal) = _checkLoan(loanId);
 
+        /// Cache the Loan Token Price
+        address loanTokenPoolAddress = underlyingToLoanPool[loanParamsLocal.loanToken];
+        require(loanTokenPoolAddress != address(0), "Invalid loan token pool address");
+
+        TempLoanPoolBalance memory loanPoolBalance =
+            TempLoanPoolBalance({
+                initialLoanTokenSupply: ILoanTokenModules(loanTokenPoolAddress).totalSupply(),
+                latestLoanTokenSupply: 0
+            });
```

Comment:
same here. could reuse the same modifier. please avoid copying the same code to different functions. always use internal functions or modifiers for code used in multiple places.

Reply:
Updated to modifier